airbyte/_processors/sql/snowflakecortex.py

```diff
@@ -0,0 +1,257 @@
+    ) -> None:
+        """Custom initialization: Initialize type_converter with vector_length."""
+        self._catalog = catalog
+        self._vector_length = vector_length
+        self._engine: Engine | None = None
+        self._connection_to_reuse: Connection | None = None
+        # call base call to do necessary initialization
+        RecordProcessor.__init__(self, cache=cache, catalog_manager=None)
+        self._ensure_schema_exists()
+        self.file_writer = file_writer or self.file_writer_class(cache)
+        self.type_converter = SnowflakeCortexTypeConverter(vector_length=vector_length)
+        self._cached_table_definitions: dict[str, sqlalchemy.Table] = {}
+
+    @overrides
+    def process_airbyte_messages(
+        self,
+        messages: Iterable[AirbyteMessage],
+        write_strategy: WriteStrategy,
+    ) -> None:
+        """Process a stream of Airbyte messages."""
+        if not isinstance(write_strategy, WriteStrategy):
+            raise exc.AirbyteInternalError(
+                message="Invalid `write_strategy` argument. Expected instance of WriteStrategy.",
+                context={"write_strategy": write_strategy},
+            )
+
+        if not self._catalog:
+            raise exc.AirbyteLibInternalError(
+                message="Catalog should exist but does not.",
+            )
+
+        stream_schemas: dict[str, dict] = {}
+
+        # Process messages, writing to batches as we go
+        for message in messages:
+            if message.type is Type.RECORD:
+                record_msg = cast(AirbyteRecordMessage, message.record)
+                stream_name = record_msg.stream
+
+                if stream_name not in stream_schemas:
+                    stream_schemas[stream_name] = self._get_json_schema_from_catalog(
+                        stream_name=stream_name
+                    )
+
+                self.process_record_message(
+                    record_msg,
+                    stream_schema=stream_schemas[stream_name],
+                )
+
+            elif message.type is Type.STATE:
+                state_msg = cast(AirbyteStateMessage, message.state)
+                if state_msg.type in [AirbyteStateType.GLOBAL, AirbyteStateType.LEGACY]:
+                    self._pending_state_messages[f"_{state_msg.type}"].append(state_msg)
+                else:
+                    stream_state = cast(AirbyteStreamState, state_msg.stream)
+                    # to-do: remove hardcoding below
+                    stream_name = "myteststream"  # stream_state.stream_descriptor.name
+                    self._pending_state_messages[stream_name].append(state_msg)
+
+            else:
+                # Ignore unexpected or unhandled message types:
+                # Type.LOG, Type.TRACE, Type.CONTROL, etc.
+                pass
+
+        # in base class, following reads from expected streams, hence overriding
+        for stream_name in stream_schemas:
```

Comment:
this relies on self.expected_streams in base class, hence calling the  write_stream_data method directly